run-tests.php

```diff
@@ -1377,6 +1388,19 @@ function run_all_tests_parallel($test_files, $env, $redir_tested) {
 		}
 		$workerProcs[$i] = $proc;
 
+		if ($useSockets) {
+			$workerSock = stream_socket_accept($listenSock, 5);
+			if ($workerSock === FALSE) {
+				kill_children($workerProcs);
+				error("Failed to accept connection from worker $i");
+			}
+			$workerInput = $workerSock;
+			$workerOutput = $workerSock;
+		} else {
+			$workerInput = $pipes[0];
```

Comment:
Would it make sense to use sockets on all platforms to reduce code?

Reply:
Maybe. Not using sockets means a simpler codepath less likely to face problems, though.